thinkhazard_processing/models.py

```diff
@@ -1,63 +1,117 @@
+    id = Column(Integer, primary_key=True)
+    # the code for the hazard unit
+    # eg: cm, dm, m, km/h, PGA-g, PGA-gal, SA-g
+    code = Column(String(7), nullable=False, unique=True)
+
+
+class HazardSet(Base):
+    __tablename__ = 'hazardset'
+    __table_args__ = {u'schema': 'processing'}
+
+    # id is the string id common to the 3 layers,
+    # as reported by geonode ("hazard_set" field), eg: "EQ-PA"
+    id = Column(String, primary_key=True)
+
+    # a hazardset is related to a hazard type:
+    hazardtype_id = Column(Integer,
+                           ForeignKey('datamart.enum_hazardtype.id'),
+                           nullable=False)
 
-    # hazard_set is the ID common to the 3 layers,
-    # as reported by geonode (hazard_set field)
-    hazard_set_id = Column(String(1024), primary_key=True)  # eg: "EQ-PA"
-    # rating 1 is the rating coming from the metadata
-    rating1 = Column(Integer)
-    # rating 2 is the rating coming from the data (quality of survey)
-    rating2 = Column(Integer)
-    # "is_global" is set to true when bounds = -180/-90/180/90
-    # false otherwise
-    is_global = Column(Boolean)
-    # "processed" is set to true by the app
-    # when all 3 datasets have been treated
-    # it maintains the application state
-    processed = Column(Boolean, default=False)
+    # "local" is set to false when bounds = -180/-90/180/90
+    # this value comes from the linked layers
+    local = Column(Boolean)
+    # date the data was last updated (defaults to created):
+    # this value comes from the linked layers
+    data_lastupdated_date = Column(Date, nullable=False)
+    # date the metadata was last updated (defaults to created):
+    # this value comes from the linked layers
+    metadata_lastupdated_date = Column(Date, nullable=False)
+    # quality rating for the hazard calculation method
+    # ranges from 0 (bad) to 10 (excellent).
+    # this value comes from the linked layers
+    calculation_method_quality = Column(Integer)
+    # quality rating for the study
+    # ranges from 0 (bad) to 2 (excellent)
+    # this value comes from the linked layers
+    scientific_quality = Column(Integer)
+
+    # processing steps:
+    # a hazardset starts incomplete.
+    # then it becomes complete, which means:
+    #   * all layers have been downloaded
+    #   * the date, quality, etc fields of the hazardset has been updated
+    complete = Column(Boolean, nullable=False, default=False)
+    # finally it is processed:
+    processed = Column(Boolean, nullable=False, default=False)
 
 
 class Layer(Base):
     __tablename__ = 'layer'
     __table_args__ = {u'schema': 'processing'}
+    # a layer is identified by it's return_period and hazard_set:
+    hazardset_id = Column(String, ForeignKey('processing.hazardset.id'),
+                          primary_key=True)
+    # the return period is typically 100, 475, 2475 years but it can vary
+    return_period = Column(Integer, primary_key=True)
```

Comment:
This will probably be changed later.
With @arnaud-morvan, we think that the return_period is informative, it should not be used as criteria for the processing script.